src/snowflake/connector/result_batch.py

```diff
@@ -207,13 +210,13 @@ def __init__(
         rowcount: int,
         chunk_headers: Optional[Dict[str, str]],
         remote_chunk_info: Optional["RemoteChunkInfo"],
-        column_names: Sequence[str],
+        schema: List["ResultMetadata"],
```

Comment:
You usually want to be specific when you define a return type, or the type of a variable.
But also want to be as generic as possible when defining a function's arguments.
For this reason we should be using `Sequence` for arguments. Python also advises the same practice: https://docs.python.org/3/library/typing.html#typing.List